pkg/pillar/docs/zedkube.md

```diff
@@ -0,0 +1,134 @@
+# Clustered eve nodes (aka zedkube)
+
+## Overview
+
+## Components
+
+### kubenodeop
+
+kubenodeop handles cordoning, uncordoning, and draining of clustered eve-os nodes.  Any given node could be hosting
+one or more longhorn volume replicas and thus could be the rebuild source for other node replicas.  A drain operation should be
+performed before any Node Operation / Node Command which can cause an extended outage of a node such as a reboot, shutdown, reset.
+kubenodeop handles NodeDrainRequest objects which zedkube subscribes to, initiates the drain, and publishes NodeDrainStatus objects.
+
+### Drain PubSub setup (node reboot/shutdown)
+
+1. zedagent/handlenodedrain.go:initNodeDrainPubSub()
+    subscribes to NodeDrainStatus from zedkube
+    creates publication of NodeDrainRequest
+1. nodeagent/handlenodedrain.go:initNodeDrainPubSub()
+    subscribe to NodeDrainStatus from zedkube
+
+### Drain Request path (node reboot/shutdown)
+
+1. zedagent/parseconfig.go:scheduleDeviceOperation()
+    call GetNodeDrainStatus() to determine if system supports drain
+        systems not HV=kubevirt will return NOTSUPPORTED
+        HV=kubevirt will return:
+            NOTSUPPORTED if in single node.
+            NOTREQUESTED if in cluster mode
+    NodeDrainStatus == NOTREQUESTED:
+        - kubeapi.RequestNodeDrain
+        - then set appropriate reboot or shutdown cmd deferred state in zedagentContext struct
+    NodeDrainStatus == REQUESTED: then set appropriate reboot or shutdown cmd deferred state in zedagentContext struct
+
+### Drain Status Handler (node reboot/shutdown)
+
+1. zedagent/handlenodedrain.go:handleNodeDrainStatusImpl()
+    if drain status FAILEDCORDON or FAILEDDRAIN, unpublish NodeDrainRequest
+1. zedagent/zedagent.go:handleNodeAgentStatusImpl()
+    drainInProgress = getConfigContext.drainInProgress
```

Comment:
this looks weird in the rendered file: https://github.com/lf-edge/eve/blob/f8ffca4fb2beb8698f14ed50ade2b40b84a96e5f/pkg/pillar/docs/zedkube.md

Reply:
Good point, I'll see if I can clean it up a bit

Reply:
Is this better?

Reply:
Yes, thx!